packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1274 @@
+/* eslint-disable sonarjs/cognitive-complexity */
+/* eslint-disable fp/no-delete */
+/* eslint-disable functional/immutable-data */
+import { fail } from "assert";
+import { generateMock } from "@anatine/zod-mock";
+import {
+  getMockAgreement,
+  getMockCertifiedTenantAttribute,
+  getMockDeclaredTenantAttribute,
+  getMockDescriptorPublished,
+  getMockEService,
+  getMockEServiceAttribute,
+  getMockTenant,
+  getMockVerifiedTenantAttribute,
+  getRandomAuthData,
+  StoredEvent,
+} from "pagopa-interop-commons-test";
+import {
+  Agreement,
+  AgreementAddedV1,
+  AgreementConsumerDocumentAddedV1,
+  AgreementDocumentId,
+  AgreementId,
+  AgreementV1,
+  AttributeId,
+  Descriptor,
+  DescriptorId,
+  Document,
+  EService,
+  EServiceId,
+  StampsV1,
+  TenantId,
+  agreementState,
+  descriptorState,
+  generateId,
+  protobufDecoder,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { afterAll, beforeAll, describe, expect, it, vi } from "vitest";
+import { v4 as uuidv4 } from "uuid";
+import { FileManagerError, userRoles } from "pagopa-interop-commons";
+import {
+  toAgreementStateV1,
+  toAgreementV1,
+  toStampV1,
+} from "../src/model/domain/toEvent.js";
+import { config } from "../src/utilities/config.js";
+import {
+  agreementAlreadyExists,
+  agreementNotFound,
+  agreementNotInExpectedState,
+  descriptorNotFound,
+  eServiceNotFound,
+  missingCertifiedAttributesError,
+  noNewerDescriptor,
+  operationNotAllowed,
+  publishedDescriptorNotFound,
+  tenantIdNotFound,
+  unexpectedVersionFormat,
+} from "../src/model/domain/errors.js";
+import {
+  addOneAgreement,
+  addOneEService,
+  addOneTenant,
+  readAgreementEventByVersion,
+  readLastAgreementEvent,
+} from "./utils.js";
+import {
+  eservices,
+  tenants,
+  postgresDB,
+  agreementService,
+  agreements,
+  fileManager,
+} from "./agreementService.integration.test.js";
+
+export function testUpgradeAgreement(): void {
```

Comment:
After you rebased I made another slight change in https://github.com/pagopa/interop-be-monorepo/pull/377 - returning the describe result to make sure it's actually equivalent to calling describe in the main tests file: https://github.com/pagopa/interop-be-monorepo/commit/f6322ea9fa07070dd84d473a29da930e91637b1c

Could you integrate it here as well? 🙏

Reply:
👍🏻